src/main/java/com/sap/sgs/phosphor/fosstars/data/github/FirstCommit.java

```diff
@@ -29,21 +29,18 @@
   /**
    * Initializes a data provider.
    *
-   * @param where A GitHub organization of user name.
-   * @param name A name of a repository.
    * @param github An interface to the GitHub API.
    */
-  public FirstCommit(String where, String name, GitHub github) {
-    super(where, name, github);
+  public FirstCommit(GitHub github) {
+    super(github);
   }
 
   @Override
-  public FirstCommit update(ValueSet values) throws IOException {
-    Objects.requireNonNull(values, "Hey! Values can't be null!");
+  protected FirstCommit doUpdate(GitHubProject project, ValueSet values) throws IOException {
     logger.info("Figuring out when the first commit was done ...");
 
-    Value<Date> firstCommit = firstCommitDate();
-    cache().put(url, firstCommit);
+    Value<Date> firstCommit = firstCommitDate(project);
+    cache.put(project, firstCommit);
```

Comment:
Maybe it would be better to stick to one thing cache() or cache.

Reply:
I wanted to use the `cache` field inside the data providers (like here). I'll check if I missed something.

Reply:
ok